### DISCUSSION TITLE
yarn patch --update for new versions?

### DISCUSSION THREAD

**Discussion by cellog** [yarnpkg/berry#6460] [Q&A]
I've noticed that when upgrading a dependency, the patch stops being applied, but running `yarn patch --update` with no changes generates an empty patch file that fails the installation. This seems like a pretty clear bug, can you all confirm before I report it? At the very least, it should error if I'm doing things wrong, instead of generating an invalid state, no?

**Comment by snowystinger** [comment]
Mostly chiming in so I get updates about this, however, I thought update wasn't for if I had updated the underlying version, but rather for updating the patch itself. However, I wasn't able to get that working either.
What is the purpose of --update?
